TTHAnalysis/python/plotter/susy-sos/functionsSF.cc

```diff
@@ -4,770 +4,691 @@
-
-		assert (res>0 && "*** Warning we have a negative (or zero) Trigger efficiency ***");
-		return res;
-	} 
-	return 1;
-} 
-
-
-// LEPTON SCALE FACTORS FULLSIM
-// -------------------------------------------------------------
-
-// electrons
-TFile* f_elSF_looseToTight_barrel_16      = new TFile(DATA_SF+"/sos_lepton_SF/el_SOS_barrel_36invfb.root", "read");
-TFile* f_elSF_looseToTight_endcap_16      = new TFile(DATA_SF+"/sos_lepton_SF/el_SOS_endcap_36invfb.root", "read");
-TGraphAsymmErrors* h_elSF_looseToTight_barrel_16      = (TGraphAsymmErrors*) f_elSF_looseToTight_barrel_16->Get("ratio");
-TGraphAsymmErrors* h_elSF_looseToTight_endcap_16      = (TGraphAsymmErrors*) f_elSF_looseToTight_endcap_16->Get("ratio");
-
-int getBinElectronLoose_16(float pt){
-	if     (pt >  5.0 && pt <= 12.5) return 0;
-	else if(pt > 12.5 && pt <= 16.0) return 1;
-	else if(pt > 16.0 && pt <= 20.0) return 2;
-	else if(pt > 20.0 && pt <= 25.0) return 3;
-	else if(pt > 25.0              ) return 4;
-	else {
-		assert(0);
+	// Low MET triggers
+	else{
+		// Mu leg
+		float mu1_Data	= h_trigEff_mumuMET_muleg_Data[_year]->GetBinContent(h_trigEff_mumuMET_muleg_Data[_year]->GetXaxis()->FindBin(_pt1),
+																			h_trigEff_mumuMET_muleg_Data[_year]->GetYaxis()->FindBin(eta1));
+		float mu1_MC	= h_trigEff_mumuMET_muleg_MC[_year]->GetBinContent(h_trigEff_mumuMET_muleg_MC[_year]->GetXaxis()->FindBin(_pt1),
+																			h_trigEff_mumuMET_muleg_MC[_year]->GetYaxis()->FindBin(eta1));
+		float mu2_Data	= h_trigEff_mumuMET_muleg_Data[_year]->GetBinContent(h_trigEff_mumuMET_muleg_Data[_year]->GetXaxis()->FindBin(_pt2),
+																			h_trigEff_mumuMET_muleg_Data[_year]->GetYaxis()->FindBin(eta2));
+		float mu2_MC	= h_trigEff_mumuMET_muleg_MC[_year]->GetBinContent(h_trigEff_mumuMET_muleg_MC[_year]->GetXaxis()->FindBin(_pt2),
+																			h_trigEff_mumuMET_muleg_MC[_year]->GetYaxis()->FindBin(eta2));
+
+		// Met leg
+		float met_Data	= h_trigEff_mumuMET_metleg_Data[_year]->GetBinContent(h_trigEff_mumuMET_metleg_Data[_year]->GetXaxis() ->FindBin(met),
+																			h_trigEff_mumuMET_metleg_Data[_year]->GetYaxis()->FindBin(met_corr));
+		float met_MC	= h_trigEff_mumuMET_metleg_MC[_year]->GetBinContent(h_trigEff_mumuMET_metleg_MC[_year]->GetXaxis() ->FindBin(met),
+																			h_trigEff_mumuMET_metleg_MC[_year]->GetYaxis()->FindBin(met_corr));
+
+		// Putting everything together
+		if(_year == 2016){
+			// DCA efficiency included in the mu leg => Divide by it to avoid double counting
+			eff_Data	= mu1_Data * mu2_Data / dcaDz_Data[_year] * mass_Data * met_Data;
+			eff_MC		= mu1_MC * mu2_MC / dcaDz_MC[_year] * mass_MC * met_MC;
+		} else {
+			// DZ efficiency computed separately from mu leg => Multiply by it
+			eff_Data	= mu1_Data * mu2_Data * dcaDz_Data[_year] * mass_Data * met_Data;
+			eff_MC		= mu1_MC * mu2_MC * dcaDz_MC[_year] * mass_MC * met_MC;
+		}
+		SF = (eff_MC == 0.0) ? 0.0 : eff_Data / eff_MC;
```

Comment:
Is this the correct way to treat zeroes in the denominator of the SF?

Reply:
Negative of zero efficiencies or scale factors should not occur. A large message has been added in [26dcdb6](https://github.com/cms-susy-soft-leptons/cmgtools-lite/pull/1/commits/26dcdb691f3cd20324f6d36d9b52ab5d20cb4b19) to spot such cases.